src/sinks/file/streaming_sink/compat.rs

```diff
@@ -0,0 +1,66 @@
+use super::StreamingSink;
```

Comment:
I think it makes sense to just expose a struct that handles all of this instead of a bunch of functions for now. Lets try to keep it simple.

Reply:
What do you mean? Something like `struct Compat`? Sorry, I don't understand what you're proposing here.